core/src/main/java/io/ddf/analytics/ABinningHandler.java

```diff
@@ -22,21 +30,192 @@ public ABinningHandler(DDF theDDF) {
   public abstract List<AStatisticsSupporter.HistogramBin> getVectorApproxHistogram(String column, int numBins)
           throws DDFException;
 
-  public DDF binning(String column, String binningType, int numBins, double[] breaks, boolean includeLowest,
-      boolean right) throws DDFException {
+  public DDF binning(String column, String binningType, int numBins, double[] breaks, boolean toLabels, boolean includeLowest,
+      boolean right, int precision) throws DDFException {
+
+    Schema.Column colMeta = this.getDDF().getColumn(column);
+
+    BinningType binType = BinningType.get(binningType);
+
+    double[] theBreaks = breaks;
+
+    switch(binType) {
+      case CUSTOM:
+        if (breaks == null) throw new DDFException("Please enter valid break points");
+        // check for monotonic
+        Arrays.sort(theBreaks);
+        if (!Arrays.equals(theBreaks, breaks)) throw new DDFException("Please enter increasing breaks");
+        break;
+      case EQUALFREQ:
+        if (numBins < 2) throw new DDFException("Number of bins cannot be smaller than 2");
+        theBreaks = getQuantilesFromNumBins(colMeta.getName(), numBins);
+        includeLowest = true;
+        break;
+      case EQUALINTERVAL:
+        if (numBins < 2) throw new DDFException("Number of bins cannot be smaller than 2");
+        theBreaks = getIntervalsFromNumBins(colMeta.getName(), numBins, right);
+        includeLowest = true;
+        break;
+      default:throw new DDFException(String.format("Binning type %s is not supported", binningType));
+    }
+
+    // Check for uniqueness
+    List<Double> listBreaks = Arrays.asList(ArrayUtils.toObject(theBreaks));
+    if (new HashSet<>(listBreaks).size() < theBreaks.length) {
```

Comment:
One java8 way to do this is: `if (!Arrays.stream(theBreaks).allMatch(new HashSet<>()::add)) {`